api/src/last-submission/last-submission.controller.ts

```diff
@@ -0,0 +1,49 @@
+import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common'
+import { LastSubmissionService } from './last-submission.service'
+import { LastSubmissionDto } from './dto/last-submission.dto'
+
+@Controller({
+  path: 'last-submission',
+  version: '1'
+})
+export class LastSubmissionController {
+  constructor(private readonly lastSubmissionService: LastSubmissionService) {}
+
+  @Post()
+  async create(@Body() lastSubmissionDto: LastSubmissionDto) {
+    return this.lastSubmissionService.create(lastSubmissionDto)
+  }
+
+  @Get()
+  async findAll() {
+    return this.lastSubmissionService.findAll({})
+  }
+
+  @Get(':id')
+  async findOne(@Param('id') id: string) {
+    return this.lastSubmissionService.findOne({ id: Number(id) })
+  }
+
+  @Get(':hash/latest')
```

Comment:
The API endpoint is called `last-submission`. I think we don't have to specify the endpoint additionally with `latest`. What do you think?

```suggestion
  @Get(':hash')
```